### ISSUE TITLE
Compilation broken after #31 (log4tango removal) was merged

### ISSUE THREAD

**Issue by mliszcz** [tango-controls/TangoSourceDistribution#32]
```
$ g++ -I/usr/local/include/tango main.cpp $(pkg-config --libs tango); echo $?
/usr/bin/ld: cannot find -llog4tango
collect2: error: ld returned 1 exit status
1

$ pkg-config --libs tango
-L/usr/local/lib -ltango -lomniDynamic4 -lCOS4 -lomniORB4 -lomnithread -llog4tango -lzmq
```

log4tango dependency must be removed from tango pkg-config:
https://github.com/tango-controls/TangoSourceDistribution/blob/f409a5abc8777cd747ff0301d020706db5628a3c/assets/lib/cpp/tango.pc.in#L6-L10

I'll send a PR.

**Comment by mliszcz** [comment]
The issue is much more complex than I initially tought.

There are 29 occurences of `llog4tango` across all repositories:
https://github.com/search?p=3&q=org%3Atango-controls+llog4tango&type=Code

And there are 458 occurences of `log4tango`.
https://github.com/search?q=org%3Atango-controls+log4tango&type=Code

For sure we need to fix the first 29 occurences, because these are linker flags.

I'll start submitting the PRs for various repositories, but if you see a better way to approach this, please let me know.

**Comment by bourtemb** [comment]
> There are 29 occurences of `llog4tango` across all repositories:
> https://github.com/search?p=3&q=org%3Atango-controls+llog4tango&type=Code
> 
> And there are 458 occurences of `log4tango`.
> https://github.com/search?q=org%3Atango-controls+log4tango&type=Code

The recent merge of liblog4tango and libtango will not have any impact on the majority of these occurrences.

> For sure we need to fix the first 29 occurences, because these are linker flags.

Even in these 29 occurrences, there are some where there is currently not impact on the generated code.
For instance, the cpp_test_ds Makefile is no longer used...
But it's true we could take this opportunity to do some cleanup and remove the Makefiles which are no longer used.

> 
> I'll start submitting the PRs for various repositories, but if you see a better way to approach this, please let me know.

I think we should update on a case by case basis.
We should first focus on the software distributed and compiled with the distribution.
We can then submit PRs to the different projects to fix all of this...

There might be some changes needed too in the compilation include flags because the includes are now installed in _prefix_/include/tango

**Comment by mliszcz** [comment]
I've already removed logtango:
* from Pogo's cmake templates on windows
* and from Makefiles in source distribution

Is this enough to release the source distribution?  I think we could remove Makefiles from other projects only after we migrate source distribution from autotools to cmake.

**Comment by bourtemb** [comment]
> I've already removed logtango:
> 
>     * from Pogo's cmake templates on windows
> 
>     * and from Makefiles in source distribution
> 
> 
> Is this enough to release the source distribution?

It's probably enough indeed for this topic, once we will have merged #33 . There are still a few updates to be done before the release, like getting the latest release versions from each SW component and #36 to be reviewed and merged.

> I think we could remove Makefiles from other projects only after we migrate source distribution from autotools to cmake.

We can do that after migrating to CMake indeed. No emergency here.

**Comment by bourtemb** [comment]
Fixed by #33
